### PR TITLE
GIX-2117: Add cketh services

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Add services to load the token and accounts of ICRC tokens.

# Changes

* New icrc accounts services `loadIcrcToken`, `loadIcrcAccount` and `loadIcrcBalances`

# Tests

* Test new services.

# Todos

- [ ] Add entry to changelog (if necessary).

Not yet necessary.

### CODE REVIEW

**Comment by lmuntaner** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3861** <a href="https://app.graphite.dev/github/pr/dfinity/nns-dapp/3861?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dfinity/nns-dapp/3861?utm_source=stack-comment).

**Comment by lmuntaner** [comment]
@peterpeterparker @dskloetd please review

File: frontend/src/lib/services/icrc-accounts.services.ts [hunk 1]
```diff
@@ -1,19 +1,158 @@
...
+        ledgerCanisterId,
+        certified,
+      }),
+    onLoad: async ({ response: account, certified }) =>
+      icrcAccountsStore.set({
+        accounts: { accounts: [account], certified },
+        universeId: ledgerCanisterId,
+      }),
+    onError: ({ error: err, certified }) => {
+      if (certified && notForceCallStrategy()) {
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/lib/services/icrc-accounts.services.ts:116]: Yes, true. Let me rename. I started with one idea and kind of changed along the way.

File: frontend/src/tests/lib/services/icrc-accounts.services.spec.ts [hunk 2]
```diff
@@ -13,4 +57,116 @@ describe("icrc-accounts-services", () => {
...
+      });
+    });
+
+    it("loads token from api into store with query", async () => {
+      expect(get(tokensStore)[ledgerCanisterId.toText()]).toBeUndefined();
+
+      await loadIcrcToken({ ledgerCanisterId, certified: false });
+
+      expect(get(tokensStore)[ledgerCanisterId.toText()]).toEqual({
+        certified: false,
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/tests/lib/services/icrc-accounts.services.spec.ts:80]: Good point, done in both tests, that `certified: true` is not used in the calls.

File: frontend/src/lib/services/icrc-accounts.services.ts [hunk 3]
```diff
@@ -1,19 +1,158 @@
...
+        err,
+      });
+
+      // Hide unproven data
+      tokensStore.resetUniverse(ledgerCanisterId);
+    },
+  });
+};
+
+export const loadIcrcBalances = async ({
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/lib/services/icrc-accounts.services.ts]: Renamed

**Comment by lmuntaner** [comment]
@dskloetd ready for another review

File: frontend/src/tests/lib/services/icrc-accounts.services.spec.ts [hunk 4]
```diff
@@ -13,4 +58,141 @@ describe("icrc-accounts-services", () => {
...
+        token: mockToken,
+      });
+      expect(ledgerApi.queryIcrcToken).toHaveBeenCalledWith({
+        certified: false,
+        identity: mockIdentity,
+        canisterId: ledgerCanisterId,
+      });
+      expect(ledgerApi.queryIcrcToken).not.toHaveBeenCalledWith({
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/tests/lib/services/icrc-accounts.services.spec.ts]: Good point, done

File: frontend/src/tests/lib/services/icrc-accounts.services.spec.ts [hunk 5]
```diff
@@ -13,4 +58,141 @@ describe("icrc-accounts-services", () => {
...
+  describe("loadIcrcToken", () => {
+    it("loads token from api into store", async () => {
+      expect(get(tokensStore)[ledgerCanisterId.toText()]).toBeUndefined();
+
+      await loadIcrcToken({ ledgerCanisterId, certified: true });
+
+      expect(get(tokensStore)[ledgerCanisterId.toText()]).toEqual({
+        certified: true,
+        token: mockToken,
+      });
```
[the remark this answers is not in the document]
Comment (lmuntaner) [reply frontend/src/tests/lib/services/icrc-accounts.services.spec.ts:71]: done

**Comment by lmuntaner** [comment]
@dskloetd ready for another review

**Review by dskloetd — APPROVED** [review]

**Pull request merged**